### PR TITLE
[DNM] fixes mounted turrets, multi-seat speeduwagon now ride with metabuddies

### PR DESCRIPTION
**Pull request by Militaires**
one bug i need to iron out with turrets so wait on until i delet dnm title

### CODE REVIEW

File: code/datums/riding.dm [hunk 1]
```diff
@@ -264,19 +269,75 @@
...
-				if(SOUTH)
-					buckled_mob.pixel_x = 16
-					buckled_mob.pixel_y = 3
-				if(EAST)
-					buckled_mob.pixel_x = -4
-					buckled_mob.pixel_y = 30
-				if(WEST)
-					buckled_mob.pixel_x = 4
-					buckled_mob.pixel_y = -1
+			if(buckled_mob = ridden.buckled_mobs[1])
```
Comment (RemieRichards) [inline code/datums/riding.dm]: no thanks.
give ```/datum/riding``` a proc, ```/datum/riding/proc/passanger_index_to_pixel_offsets()``` and then we can generalize this to all vehicles.

```passanger_index_to_pixel_offsets()``` would return ```list("[NORTH]" = list("x" = 0, "y" = 0), "[SOUTH]" = list("x" = 0, "y" = 0), ...)```

Comment (BeeSting12) [inline code/datums/riding.dm]: Why is passenger spelled wrong there?

Comment (RemieRichards) [inline code/datums/riding.dm]: because I typed it quickly.

Comment (Militaires) [inline code/datums/riding.dm]: why cant all vehicles use the easier and less expensive method

Comment (Militaires) [inline code/datums/riding.dm]: @remierichards this also has to change layers sometimes your method sounds needless really this is way simpler and probably less expensive or is it @cyberboss

**Comment by Robustin** [comment]
Hmm, just like my vehicle update down to the pixel offsets and double overlay for the speedwagon, hmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmm

Because you couldn't wait 24 hours for me to update my PR?

File: code/datums/riding.dm (hunk 1 above)
Comment (RemieRichards) [inline code/datums/riding.dm]: @Militaires No, you'll do as I say, speed is not the concern here, Maintainability is, and I'm the maintainer. Chop chop.

**Comment by lzimann** [comment]
Tell me whenever you come back and if you want this reopened.

File: code/datums/riding.dm (hunk 1 above)
Comment (Militaires) [inline code/datums/riding.dm]: @RemieRichards in what world is your needless proc more maintainable than a switch and some simple logic

**Comment by Militaires** [comment]
@Robustin except i didnt define needless proc and fail to use the vars already given to me

**Comment by Militaires** [comment]
@KorPhaeron please re-open this

File: code/datums/riding.dm (hunk 1 above)
Comment (duncathan) [inline code/datums/riding.dm]: holy memes separating stuff into a proc is almost always more maintainable are you dumb

**Comment by KorPhaeron** [comment]
I cant, button is greyed out

**Pull request closed without merging**